### ISSUE TITLE
DEX integration

### ISSUE THREAD

**Issue by Kisgus** [openbot-tech/core#13]
## Userstory

As a user I want to be able to run my custom strategies via a DEX, so that the deployed bot can tap into the liquidity and markets of DEXs

## Acceptance Criteria

Openbot is integrated and works with

- [ ] [DDEX](https://docs.ddex.io/)

- [ ] [Radar Relay](https://developers.radarrelay.com/)

**Comment by Kisgus** [comment]
Proposed by Ben from Archon that we use this JS implementation https://github.com/RadarTech/sdk-bot-example

**Comment by PumpkingWok** [comment]
Hi @Kisgus,
I can give a help in my free time if no one is already working on this issue.
Thanks in advance.

**Comment by Kisgus** [comment]
Hey @PumpkingWok that would be great. Right now @rootkc is looking into RadarRelay integration. Have you tried to integrate with that one before?

**Comment by PumpkingWok** [comment]
Hi @Kisgus,
I have never tried to integrate a Dex/RadarRelay into a trading bot, only classic exchanges via API or websockets, in many languages, for personal trading strategy. 
In fact i would like to try to help you also for trying it for the first time.
No worries if someone already is working in it or you are in a hurry.
Thanks in advance.

**Comment by Kisgus** [comment]
Sounds great @PumpkingWok, we would love your contribution and get you onboard! How about you start looking into integrating DDEX? https://docs.ddex.io/

I have helped integrate 0x v1 before, but not this Radar Relay version built on v2, this is what @rootkc is working on. A blocker we have right now is where to get the market-data, we see other implementations get those from for-example coinmarket cap which we are not happy about that.

**Comment by PumpkingWok** [comment]
Hi @Kisgus,
Have you looked into https://coinhub.diadata.org/ ?
It is a DIAData project https://github.com/diadata-org/diadata, i have developed many exchanges scraper for them, written in golang.
I'm going to try to implement ddex this week.

**Comment by PumpkingWok** [comment]
Hi @Kisgus,
I have looked into DDEX documentation, I noticed that they moved to API v3 since January 2019.
They provide a Hydro API js SDK, https://github.com/HydroProtocol/sdk
Would you like that I implement it ?

**Comment by Kisgus** [comment]
Yes that would be great @PumpkingWok, do you think it is possible to connect it to the broker websocket connection?
